File: Codigos/Codigos/JAVA/WorkSpace C2/Agenda/src/Model/pessoa.java
package Model;

public class pessoa {

	private String nome;
	private data nascimento;
	private double altura;
	
	public pessoa(String nome, data nascimento, double altura) {
		this.nome = nome;
		this.nascimento = nascimento;
		this.altura = altura;
	}
	
	public String getNome() {
		return nome;
	}
	
	public void setNome(String nome) {
		this.nome = nome;
	}
	
	public data getNascimento() {
		return nascimento;
	}
	
	public void setNascimento(data nascimento) {
		this.nascimento = nascimento;
	}
	
	public double getAltura() {
		return altura;
	}
	
	public void setAltura(double altura) {
		this.altura = altura;
	}
	
	public int getIdade(data hoje) {
		return this.nascimento.diferencaAno(hoje);
	}
	
	public boolean equals (String nome) {
		return this.nome.toUpperCase().equals(nome.toUpperCase());
	}
	
	public String toString() {
		return "Nome: " + this.nome + "\n" + "Data de Nascimento: " + this.nascimento + "\n" + "Altura: " + this.altura + " metros";
	}
	
}

File: Codigos/Codigos/C++/C2/Listas e Trabalhos/LISTA 28-06-18/20180628/pessoa.cpp
#include "pessoa.h"

pessoa::pessoa(){
    this->nome = "";
    this->nascimento = NULL;
    this->altura = 0;
}

pessoa::pessoa(string nome, data *nascimento, float altura){
    this->nome = nome;
    this->nascimento = nascimento;
    this->altura = altura;
}

string pessoa::getNome(){
    return this->nome;
}

data* pessoa::getNascimento(){
    return this->nascimento;
}

float pessoa::getAltura(){
    return this->altura;
}

void pessoa::setNome(string nome){
    this->nome = nome;
}

void pessoa::setNascimento(data *Data){
    this->nascimento = Data;
}

void pessoa::setAltura(float altura){
    this->altura = altura;
}

void pessoa::idadeCalc(data* nascimento){
    this->idade = (2018 - nascimento->getAno());
}

void pessoa::mostrarPessoa(data* nascimento){
    cout << "Nome: " << this->nome << endl;
    cout << "Data de Nascimento: " << nascimento->getAno() << endl;
    cout << "Altura: " << this->altura << " metros" << endl;
    cout << "Idade: " << this->idade << " anos" << endl;
}

pessoa::~pessoa(){
    cout << "Destruindo o objeto: ..." << endl;
}
